### PR TITLE
Only Refresh next_occurrence When Not Expired

### PR DESCRIPTION
**Pull request by tylerneu**
If an `RRule` is updated _after_ `next_occurrence` but _before_ the handler runs, `next_occurrence` will be updated without the handler being executed. 

These changes only refresh the occurrence when it isn't expired allowing the handler to run and update the next occurrence.

### CODE REVIEW

File: ambition_utils/rrule/forms.py [hunk 1]
```diff
@@ -249,13 +249,33 @@ def save(self, **kwargs):
...
+            original_rrule_params = sorted(
+                [str(value) for value in rrule_model.rrule_params.values()]
+            )
+
+            form_exclusion_params = sorted(
+                [str(value) for value in (self.cleaned_data.get('rrule_exclusion') or {}).values()]
+            )
+            original_exclusion_params = sorted(
+                [str(value) for value in (rrule_model.rrule_exclusion_params or {}).values()]
+            )
```
Comment (tylerneu) [inline ambition_utils/rrule/forms.py:271]: I considered using `sorted(..., key=str)` but was not seeing `datetime` & `FakeDatetime` converted as expected. (The existing `distart` is a string but the submitted `distort` is an object.)

I don't love this and welcome suggestions.

File: setup.cfg [hunk 2]
```diff
@@ -2,7 +2,7 @@
 max-line-length = 120
 exclude = docs,venv,env,*.egg,migrations
 max-complexity = 21
-ignore = E402,E722
+extend-ignore = E402,E722
```
Comment (tylerneu) [inline setup.cfg:5]: * https://stackoverflow.com/questions/68161741/python-black-formatter-conflict-with-rule-flake8-w503-in-vscode
* https://github.com/psf/black/blob/06ccb88bf2bd35a4dc5d591bb296b5b299d07323/docs/guides/using_black_with_other_tools.md#flake8

**Comment by monroeclinton** [comment]
Right now the task just runs and checks the `next_occurrence` in the database with no knowledge of changes. Perhaps some sort of queue should be introduced that save the rrule state for when the occurrence should run

**Comment by tylerneu** [comment]
@somewes Thank you. I really like your suggestion and have pushed changes for you to consider.

@monroeclinton I'm don't follow. Take a look at my changes and let me know if you think there is more room for improvement.

**Review by somewes — APPROVED** [review]

**Comment by somewes** [comment]
looks good to me @tylerneu 

@monroeclinton if the occurrence is already expired it means we need to run the handler. when the handler runs and generates the next occurrence, that date is based on the rrule params. in this case, if an overdue rrule object is saved with a new config, but it hasn't run its handler yet, we don't want to generate a next occurence yet. this allows the handler to run as expected, then the next occurrence is generated based on the new rrule params.

**Pull request merged**